dss/operations/util.py

```diff
@@ -0,0 +1,60 @@
+
+import os
+import traceback
+import typing
+from uuid import uuid4
+from contextlib import AbstractContextManager
+
+from cloud_blobstore import BlobStore
+
+from dss.util.aws.clients import sqs, sts
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+
+_account_id = sts.get_caller_identity()['Account']  # type: ignore
+_queue_url = "https://sqs.{}.amazonaws.com/{}/dss-operations-{}".format(os.environ['AWS_DEFAULT_REGION'],
+                                                                        _account_id,
+                                                                        os.environ['DSS_DEPLOYMENT_STAGE'])
+
+
+class CommandForwarder(AbstractContextManager):
+    """
+    Context manager for forwarding commands for Lambda execution through batch SQS queuing
+    """
+    def __init__(self):
+        self.chunk = list()
+
+    def forward(self, msg: str):
+        self.chunk.append(msg)
+        if 10 == len(self.chunk):
+            _enqueue_command_batch(self.chunk)
+            self.chunk = list()
+
+    def __exit__(self, *args, **kwargs):
+        if len(self.chunk):
+            _enqueue_command_batch(self.chunk)
+
+def map_bucket(func: typing.Callable, handle: BlobStore, bucket: str, base_pfx: str, parallelization=16):
+    """
+    Call `func` on an iterable of keys
+    func is expected to be thread safe.
+    """
+    with ThreadPoolExecutor(max_workers=parallelization) as e:
+        futures = list()
+        for pfx in "0123456789abcdef":
```

Comment:
why 0123456789abcdef?

Reply:
UUIDs are hexadecimal, so `pfx` in `0123456789abcdef` covers the bundle/file/collection namespace for `{bucket}/{type}s/{pfx}`